### PR TITLE
feat: setup code generation

### LINKED ISSUE #347: 🛠️ [TASK] : Setup code generation
### Summary

Setup [swagger_dart_code_generator](https://pub.dev/packages/swagger_dart_code_generator) for Automated Code Generation from O[penAPI Specifications](https://input-output-hk.github.io/catalyst-voices/api/cat-gateway/openapi/).

### Description

This task will automate code generation within the `catalyst_voices_services` project, leveraging the `swagger_dart_code_generator`. This will streamline the development process and ensure that our Flutter application remains in sync with the backend service's API.

#### Tasks

- [ ] **Setup `swagger_dart_code_generator`:** 
Integrate `swagger_dart_code_generator` into our project and ensure that it's properly configured to work with our current development environment.

- [ ] **Automate OpenAPI Specification Fetching:** 
Implement a mechanism (potentially using Earthly or a custom script) that automatically fetches the latest version of the OpenAPI specification from our backend service. This should ensure our generated code always matches the backend's current API.

- [ ] **Generate Code:** 
Configure the code generator to run, fetching the latest OpenAPI spec and generating the Dart code accordingly. Focus on ensuring the generated code includes all necessary models, clients, and serialization logic per our API's latest specs.

- [ ] **Organize Generated Code:**
Ensure all generated code is placed within our project structure's dedicated `generated` folder. This organization will help maintain clarity between handwritten and auto-generated code.

- [ ] **Integrate with Version Control:** Automate the inclusion of generated code in our version control system. This ensures that our generated code is always up-to-date with the latest API specifications and is versioned alongside our manual code changes.

- [ ] **Add Code Generation to Earthfile:** 
Incorporate the code generation step into our [Earthfile](https://github.com/input-output-hk/catalyst-voices/blob/main/catalyst_voices/Earthfile) to automate this process within our CI/CD pipeline. This integration should include fetching the latest API specs and generating code as part of our build process.

#### Acceptance Criteria

- `swagger_dart_code_generator` is fully integrated and configured within the `catalyst_voices_services` project.
- There's an automated process for fetching the latest OpenAPI specifications and generating Dart code accordingly.
- The generated code is saved in the `generated` folder and committed to git.
- The Earthfile includes a step for code generation, ensuring this process is integrated into our CI/CD pipeline.

### Additional Notes

- Review and test the integration thoroughly to ensure the generated code meets our project's standards and requirements.

### PR DESCRIPTION
**Pull request by coire1**
# Description

This PR introduces the Automated Code Generation for the `catalyst_voices` project using the OpenAPI specifications defined in `catalyst-gateway`. It orchestrates a set of earthly targets to build the OpenAPI specs, generate the code and validate the generated code consistencies on PRs.

## Related Issue(s)

Closes #347

### CODE REVIEW

File: catalyst_voices/Earthfile [hunk 1]
```diff
@@ -24,6 +29,34 @@ src:
...
+# and serialization logic.
+code-generator:
+    FROM +deps
+    COPY --dir pubspec.yaml lib packages web test test_driver integration_test catalyst_voices
+    COPY ../+repo-catalyst-voices-packages/repo .
+    WORKDIR /frontend/catalyst_voices/packages/catalyst_voices_services
+    COPY ../catalyst-gateway+build/doc/cat-gateway-api.json openapi/cat-gateway-api.json
+    RUN flutter pub get
+    RUN dart run build_runner build --delete-conflicting-outputs
+    SAVE ARTIFACT lib/generated
```
Comment (kukkok3) [inline catalyst_voices/Earthfile:43]: Maybe you can pass an `ARG save_locally` to this target and when it is set to true it saves the generated code locally? so you can remove the `code-generator-local` target? Something like:
```suggestion
IF [ "$save_locally" = "true" ]
        SAVE ARTIFACT  lib/generated AS LOCAL  lib/generated
    ELSE
        SAVE ARTIFACT  lib/generated
    END
```

**Review by minikin — CHANGES_REQUESTED** [review]
It fails for me when I run it locally.
Here are [logs.](https://app.warp.dev/block/lS1mCSGuZMIoAbYzikZh5s)

**Comment by coire1** [comment]
> It fails for me when I run it locally. Here are [logs.](https://app.warp.dev/block/lS1mCSGuZMIoAbYzikZh5s)

@minikin  Question to identify the issue: does it work by running `earthly --platform=linux/amd64 +code-generator-local`

File: catalyst_voices/Earthfile (hunk 1 above)
Comment (coire1) [reply catalyst_voices/Earthfile:43]: Yeah that's a good idea!

**Comment by minikin** [comment]
@coire1 this `earthly --platform=linux/amd64 +code-generator-local` command indeed works. Let's update the documentation, please.

**Review by minikin — CHANGES_REQUESTED** [review]
Please move the generated code from `catalyst_voices_models` to `catalyst_voices_services`. Cat Gateway API is a service.

**Review by minikin — APPROVED** [review]
LGTM

**Pull request merged**